Formula/bun.rb

```diff
@@ -0,0 +1,77 @@
+class Bun < Formula
+  desc "Fast all-in-one JavaScript runtime"
+  homepage "https://bun.sh"
+  license "MIT"
+
+  stable do
+    # pull from git tag to get submodules
+    url "https://github.com/oven-sh/bun.git",
+      tag:      "bun-v0.1.6",
+      revision: "fcce3ff5ac821de270e5d0094d665ec079ef205e"
+
+    # Requires unreleased zig version (0.10.0)
+    # Commit from latest fork: https://github.com/Jarred-Sumner/zig
+    resource "zig" do
+      url "https://github.com/ziglang/zig/archive/b79884eaf003ad32e800213c20da5c6b8935af34.tar.gz"
+      sha256 "f69d7525304720372520482bd54fbf62621c15b4d201465567939b15d0ba1ae5"
+    end
+  end
+
+  livecheck do
+    url :stable
+    regex(/bun-v?(\d+(?:\.\d+)+)/i)
+  end
+
+  head do
+    url "https://github.com/oven-sh/bun.git", branch: "main"
+
+    resource "zig" do
+      url "https://github.com/Jarred-Sumner/zig.git", branch: "master"
+    end
+  end
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "cmake" => :build
+  depends_on "esbuild" => :build
+  depends_on "gnu-sed" => :build
+  depends_on "go" => :build
+  depends_on "libtool" => :build
+  depends_on "llvm@13" => :build
+  depends_on "ninja" => :build
+  depends_on "node" => :build
+  depends_on "rust" => :build
+
+  def install
+    ENV["ZIG"] = buildpath/"bin/zig"
```

Comment:
Note that Zig builds with `-march=native` by default, so we want to make sure it's not doing that here. See, for example:

https://github.com/Homebrew/homebrew-core/blob/6fc08d005cfe92ca8453912306fa5704c90374b2/Formula/ncdu.rb#L29-L34